subsys/canbus/isotp/isotp.c

```diff
@@ -1068,6 +1070,9 @@ static void send_state_machine(struct isotp_send_ctx *ctx)
 				ctx->state = ISOTP_TX_WAIT_ST;
 				break;
 			}
+
+			/* Ensure FIFO style transmission of CF */
+			k_sem_take(&ctx->tx_sem, K_FOREVER);
```

Comment:
I think this should go before the `send_cf` or into the `ISOTP_TX_SEND_FF` state as the first frame is considered the frame with sequence number 0. No CF should be sent before FF.

Reply:
This is handled by the ISO-TP state machine. Once the first frame (FF) is queued, it will enter state `ISOTP_TX_WAIT_FC` and wait for the receiver to acknowledge (by sending a flow control frame (FC)) before sending the first continuation frame (CF).

Reply:
Indeed, you are correct.